zhaquirks/tuya/builder/__init__.py

```diff
@@ -0,0 +1,493 @@
+
+        This method allows exposing a switch entity in Home Assistant.
+        """
+        self.tuya_dp_attribute(
+            dp_id=dp_id,
+            attribute_name=attribute_name,
+            type=t.Bool,
+            access=foundation.ZCLAttributeAccess.Read
+            | foundation.ZCLAttributeAccess.Write,
+        )
+        self.switch(
+            attribute_name=attribute_name,
+            cluster_id=TUYA_CLUSTER_ID,
+            endpoint_id=endpoint_id,
+            force_inverted=force_inverted,
+            invert_attribute_name=invert_attribute_name,
+            off_value=off_value,
+            on_value=on_value,
+            entity_platform=entity_platform,
+            entity_type=entity_type,
+            initially_disabled=initially_disabled,
+            attribute_initialized_from_cache=attribute_initialized_from_cache,
+            translation_key=translation_key,
+            fallback_name=fallback_name,
+        )
+        return self
+
+    def tuya_enum(
+        self,
+        dp_id: int,
+        attribute_name: str,
+        enum_class: type[Enum],
+        access: foundation.ZCLAttributeAccess = foundation.ZCLAttributeAccess.Read
+        | foundation.ZCLAttributeAccess.Write,
+        endpoint_id: int = 1,
+        entity_type: EntityType = EntityType.CONFIG,
+        entity_platform: EntityPlatform = EntityPlatform.SELECT,
+        initially_disabled: bool = False,
+        attribute_initialized_from_cache: bool = True,
+        translation_key: str | None = None,
+        fallback_name: str | None = None,
+    ) -> QuirkBuilder:
+        """Add an EntityMetadata containing ZCLEnumMetadata and return self.
+
+        This method allows exposing an enum based entity in Home Assistant.
+        """
+        self.tuya_dp_attribute(
+            dp_id=dp_id,
+            attribute_name=attribute_name,
+            type=enum_class,
+            access=access,
+        )
+        self.enum(
+            attribute_name=attribute_name,
+            enum_class=enum_class,
+            cluster_id=TUYA_CLUSTER_ID,
+            endpoint_id=endpoint_id,
+            entity_type=entity_type,
+            entity_platform=entity_platform,
+            initially_disabled=initially_disabled,
+            attribute_initialized_from_cache=attribute_initialized_from_cache,
+            translation_key=translation_key,
+            fallback_name=fallback_name,
+        )
+
+        return self
+
+    def tuya_number(
+        self,
+        dp_id: int,
+        type: type,
+        attribute_name: str,
+        access: foundation.ZCLAttributeAccess = foundation.ZCLAttributeAccess.Read
+        | foundation.ZCLAttributeAccess.Write,
+        endpoint_id: int = 1,
+        min_value: float | None = None,
+        max_value: float | None = None,
+        step: float | None = None,
+        unit: str | None = None,
+        mode: str | None = None,
+        multiplier: float | None = None,
```

Comment:
https://github.com/zigpy/zigpy/pull/1487/ added `entity_type` for `number`. Let's also add it here:
```suggestion
        multiplier: float | None = None,
        entity_type: EntityType = EntityType.CONFIG,
```